src/main/java/frc/robot/subsystems/hood/Hood.java

```diff
@@ -0,0 +1,98 @@
+package frc.robot.subsystems.hood;
+
+import edu.wpi.first.units.Angle;
+import edu.wpi.first.units.MutableMeasure;
+import edu.wpi.first.units.Units;
+import edu.wpi.first.wpilibj.smartdashboard.Mechanism2d;
+import edu.wpi.first.wpilibj.smartdashboard.MechanismLigament2d;
+import edu.wpi.first.wpilibj.smartdashboard.MechanismRoot2d;
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import java.util.function.Supplier;
+import lib.Utils;
+import org.littletonrobotics.junction.AutoLogOutput;
+import org.littletonrobotics.junction.Logger;
+
+public class Hood extends SubsystemBase {
+    private static Hood INSTANCE = null;
+    private final HoodInputsAutoLogged inputs = HoodIO.inputs;
+    private final HoodIO io;
+    @AutoLogOutput private final Mechanism2d mechanism2d = new Mechanism2d(3, 3);
+    private final MechanismRoot2d root =
+            mechanism2d.getRoot(
+                    "Hood",
+                    HoodConstants.mechanism2dPose.getX(),
+                    HoodConstants.mechanism2dPose.getY());
+    private final MechanismLigament2d hood =
+            root.append(new MechanismLigament2d("Hood", HoodConstants.HoodLength, 45));
+
+    private ControlMode controlMode = ControlMode.POSITION;
+
+    /**
+     * Constructor for Hood subsystem.
+     *
+     * @param io IO of the subsystem.
+     */
+    private Hood(HoodIO io) {
+        this.io = io;
+    }
+
+    /**
+     * Gets the single instance of hoodSubsystem.
+     *
+     * @return The single instance of hoodSubsystem.
+     */
+    public static Hood getInstance() {
+        return INSTANCE;
+    }
+
+    public static void initialize(HoodIO io) {
+        INSTANCE = new Hood(io);
+    }
+
+    public MutableMeasure<Angle> getAngle() {
+        return inputs.angle;
+    }
+
+    public boolean atSetpoint() {
+        return Utils.epsilonEquals(inputs.powerSetpoint * 12, inputs.voltage.in(Units.Volts));
+    }
+
+    public Command setAngle(Supplier<MutableMeasure<Angle>> angle) {
+        controlMode = ControlMode.POSITION;
+        return runOnce(() -> inputs.angleSetpoint.mut_replace(angle.get()));
+    }
+
+    public Command setPower(Supplier<Double> power) {
+        controlMode = ControlMode.POWER;
+        return runOnce(() -> inputs.powerSetpoint = power.get());
+    }
+
+    public Command ResetAbsoluteEncoder() {
+        return setPower(() -> HoodConstants.resetPower)
```

Comment:
Add another sensor to check when to stop running the hood down (or do this check with voltage), talk with the engineering subteam to pick the right sensor.

Reply:
We're gonna use a limit switch connected directly to the falcon.

Reply:
Actually you don't need to reset the encoder yourself, the Talon can do it automatically when a limit switch is triggered.

Reply:
I talked with the mechanical team today, and why would we need to reset the absolute encoder? It's connected with a belt directly to the shooter plate, the changes of the belt jumping are basically zero.

Reply:
Yeah, I guess you're right. It won't hurt though to put a sensor if they can. Will also help for having soft limits and to reset the absolute encoder if it's getting disconnected then connected again.